contentreview-impl/impl/src/java/org/sakaiproject/contentreview/impl/turnitin/TurnitinReviewServiceImpl.java

```diff
@@ -906,6 +926,156 @@ public void addTurnitinInstructor(Map userparams) throws SubmissionException, Tr
+	        ltiProps.put("custom_report_gen_speed", extraAsnnOpts.get("report_gen_speed").toString());
+	        ltiProps.put("custom_s_view_reports", extraAsnnOpts.get("s_view_report").toString());			
+	        ltiProps.put("custom_submit_papers_to", extraAsnnOpts.get("submit_papers_to").toString());
+
+			if (extraAsnnOpts.containsKey("exclude_biblio")){
+				ltiProps.put("custom_use_biblio_exclusion", extraAsnnOpts.get("exclude_biblio").toString());
+			}
+			if (extraAsnnOpts.containsKey("exclude_quoted")){
+				ltiProps.put("custom_use_quoted_exclusion", extraAsnnOpts.get("exclude_quoted").toString());
+			}
+			
+			//adding callback url
+			String callbackUrl = serverConfigurationService.getString("serverUrl") + "/sakai-contentreview-tool-tii/tii-servlet";
+			log.debug("callbackUrl: " + callbackUrl);
+			ltiProps.put("ext_resource_tool_placement_url", serverConfigurationService.getString("serverUrl") + "/sakai-contentreview-tool-tii/tii-servlet");
+			
+			boolean result = tiiUtil.makeLTIcall(tiiUtil.BASIC_ASSIGNMENT, null, ltiProps);
+			if(!result){
+				log.error("Error making LTI call");//TODO
+				return;
+			}
+			
+			//don't create lti tool if exists
+			if(tiiId == null || ltiId == null){
+				Properties sakaiProps = new Properties();
+				String globalId = tiiUtil.getGlobalTurnitinLTIToolId();
+				sakaiProps.setProperty(LTIService.LTI_TOOL_ID,globalId);
+				
+				String custom = BasicLTIConstants.RESOURCE_LINK_ID + "=" + taskId;
+				custom += ";" + BasicLTIConstants.RESOURCE_LINK_TITLE + "=" + extraAsnnOpts.get("title").toString();
+				log.debug("Storing custom params: " + custom);
+				sakaiProps.setProperty(LTIService.LTI_CUSTOM,custom);
+				sakaiProps.setProperty(LTIService.LTI_SITE_ID,siteId);
+				
+				SecurityAdvisor advisor = new SimpleSecurityAdvisor(sessionManager.getCurrentSessionUserId(), "site.upd", "/site/!admin");
+				Object ltiContent = null;
+				try{
+					securityService.pushAdvisor(advisor);
+					ltiContent = tiiUtil.insertTIIToolContent(globalId, sakaiProps);
+				} catch(Exception e){
+					log.error("createAssignment - insertTIIToolContent: Could not push the security advisor.");
+					return;
+				} finally {
+					securityService.popAdvisor(advisor);
+				}				
+				
+				if(ltiContent == null){
+					log.error("createAssignment: Could not find LTI service.");
+					return;
+				} else if(!(ltiContent instanceof Long)){//TODO differenciate errors?
+					log.error("createAssignment: Error creating LTI stealthed tool: " + ltiContent);
+					return;
+				} else {//long if everything went fine
+					log.debug("LTI content tool id: " + ltiContent);
+					try{
+						AssignmentContentEdit ace = assignmentService.editAssignmentContent(taskId);
+						ResourcePropertiesEdit aPropertiesEdit = ace.getPropertiesEdit();
+						aPropertiesEdit.addProperty("lti_id", String.valueOf(ltiContent));
+						assignmentService.commitEdit(ace);
+					}catch(Exception e){
```

Comment:
Catching Exception is generally bad as it will catch things you aren't expecteing.